src/WebJobs.Script/Host/FunctionMetadataProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.Workers.Rpc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    internal class FunctionMetadataProvider : IFunctionMetadataProvider
+    {
+        private readonly IEnvironment _environment;
+        private readonly ILogger<FunctionMetadataProvider> _logger;
+        private IWorkerFunctionMetadataProvider _workerFunctionMetadataProvider;
+        private IHostFunctionMetadataProvider _hostFunctionMetadataProvider;
+
+        public FunctionMetadataProvider(ILogger<FunctionMetadataProvider> logger, IWorkerFunctionMetadataProvider workerFunctionMetadataProvider, IHostFunctionMetadataProvider hostFunctionMetadataProvider)
+        {
+            _logger = logger;
+            _workerFunctionMetadataProvider = workerFunctionMetadataProvider;
+            _hostFunctionMetadataProvider = hostFunctionMetadataProvider;
+            _environment = SystemEnvironment.Instance;
+        }
+
+        public ImmutableDictionary<string, ImmutableArray<string>> FunctionErrors { get; private set; }
```

Comment:
Not sure what the normal behavior of this interface is, but should this be initialized to empty immutable dictionary?

I personally prefer having a never-null approach whenever possible.